File: maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        
        int sum=0,maxsum=INT_MIN,cntmax=0;
        
        queue <TreeNode*> q;
        if(root==NULL)
            return 0;
        
       
        
        q.push(root);
        q.push(NULL);
        sum=root->val;
        int cnt=1;
        while(!q.empty())
        {
            root=q.front();
            
            q.pop();
            if(root!=NULL)
            {
                cout << root->val << endl;
                sum=sum+root->val;
                if(root->left)
                    q.push(root->left);
                if(root->right)
                    q.push(root->right);
            }
            else
            {
                cout<<sum<<" ";
                
                if(maxsum<sum)
                {
                    maxsum=sum;
                    cntmax=cnt;
                }
                cnt++;
                sum=0;
                 if(!q.empty())
                 {
                     q.push(NULL);
                 }
                
            }
        }
        
        return cntmax;
    }
};

File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
   
    public int maxLevelSum(TreeNode root) {
        
        if(root==null)
            return 0;

        Queue<TreeNode> q = new LinkedList<TreeNode>();
        q.add(root);
        int maxSum=Integer.MIN_VALUE;
        int level=0;
        int ans=0;
        while(!q.isEmpty()){
            int size = q.size();
            int currSum=0;
            level++;
            for(int i=0;i<size;i++){
                TreeNode node = q.poll();
                
                currSum+= node.val;
                if(node.left!=null){
                    q.add(node.left);
                }
                if(node.right!=null){
                    q.add(node.right);
                }
            }
            if(currSum>maxSum){
                maxSum=currSum;
                ans=level;
            }
        }
        return ans;
    }
}